### DISCUSSION TITLE
Extracting cell trajectory

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#30] [General discussion]
Hello and happy new year!
I am back with my questions!

Is there any simple way I can extract cell location at each time step?
I already know that I can extract them from the CSV files, but writing the whole data at each timestep is highly CPU-intensive. I only need to do it for one cell at a time. I guess there is something to do with hemocell.write function.

Bests,
Mojtaba

**Comment by MaxvdKolk** [comment]
There is unfortunately not a specific API that handles writing cell information of just a specific cell, or specific set of cells, to disk. You could investigate to modify `writeCellInfo_CSV` (called in `void HemoCell:writeOutput()`) to filter the written data based a given set of cell IDs of interest. 

You could either gather all cell information to a block and write the specific information from there, or scatter the desired ID to all blocks, and only write from the block that contains the cell(s) of interest.
